### PR TITLE
Merge on background

### PR DESCRIPTION
**Pull request by javitonino**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_node/src/merge/scheduler.rs [hunk 1]
```diff
@@ -41,38 +42,56 @@ use nucliadb_core::NodeResult;
...
+        let on_commit_parameters = MergeParameters {
+            max_nodes_in_merge: match std::env::var(ON_COMMIT_MAX_NODES_IN_MERGE) {
+                Ok(v) => v.parse().unwrap_or(50_000),
+                Err(_) => 50_000,
+            },
+            segments_before_merge: match std::env::var(ON_COMMIT_SEGMENTS_BEFORE_MERGE) {
+                Ok(v) => v.parse().unwrap_or(2),
+                Err(_) => 100,
+            },
+        };
```
Comment (jotare) [inline nucliadb_node/src/merge/scheduler.rs:85]: Why are we using env vars directly? Shouldn't we be using the `Settings` abstraction?

Comment (javitonino) [reply nucliadb_node/src/merge/scheduler.rs:85]: Good point. I was just copying stuff around and missed the opportunity to move it into Settings. I'll change it, thanks! :sparkles:

File: nucliadb_node/src/settings/providers.rs [hunk 2]
```diff
@@ -27,6 +27,18 @@ pub trait SettingsProvider {
...
+        if let Ok(value) = std::env::var($env) {
+            #[allow(unused_labels)]
+            'set: {
+                $(let Ok(value) = value.parse::<$parse>() else { break 'set };)*
+                $builder.$name(value);
+            }
+        }
+    };
+}
+
```
Comment (hermeGarcia) [inline nucliadb_node/src/settings/providers.rs:41]: Very cool!!

File: nucliadb_vectors/src/service/writer.rs [hunk 3]
```diff
@@ -56,26 +57,57 @@ impl Debug for VectorWriterService {
     }
 }
 
+struct DummyPreparedMerge();
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/service/writer.rs:60]: You dont need to use `()` in Rust, a zero-size type will do:
`struct DummyPrepared;`

File: nucliadb_vectors/src/service/writer.rs [hunk 4]
```diff
@@ -56,26 +57,57 @@ impl Debug for VectorWriterService {
...
+        todo!()
+    }
+
+    fn record_metrics(&self, _source: MergeSource) {
+        todo!()
+    }
+
+    fn get_metrics(&self) -> MergeMetrics {
+        todo!()
+    }
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/service/writer.rs:83]: Maybe `unreachable!` is a better fit.

**Review by hermeGarcia — APPROVED** [review]
Looks good! Just some comments

File: nucliadb_vectors2/src/data_point_provider/writer.rs [hunk 5]
```diff
@@ -245,10 +245,26 @@ impl Writer {
...
 
+        let mut new_delete_log = self.delete_log.clone();
         if let Some(oldest_age) = self.online_data_points.iter().map(|dp| dp.journal.time()).reduce(std::cmp::min) {
-            self.delete_log.prune(oldest_age);
+            new_delete_log.prune(oldest_age);
         }
 
+        let state = State {
+            delete_log: new_delete_log,
+            data_point_list: self.online_data_points.iter().map(|dp| dp.journal.id()).collect(),
```
Comment (hermeGarcia) [inline nucliadb_vectors2/src/data_point_provider/writer.rs:258]: you are iterating `self.online_data_points` twice, once for obtaining the oldest age and here. Not a big deal if we keep the number of data points low, but maybe we could organice the code in a way that only one iteration suffice?

**Pull request merged**